scripts/unbind-and-delete-all-services.sh

```diff
@@ -0,0 +1,389 @@
+  echo "dryrun:       show operations that would be run, without making any changes"
+  echo "execute:      normal operation"
+  echo "execute-prod: similar to execute, but will run against production platforms"
+  echo
+  return 1
+}
+
+#### protect against accidental destruction of production 
+prod_protect() {
+  local prod_url
+  local prod_london_url
+  local target_url
+  prod_url="https://api.cloud.service.gov.uk"
+  prod_london_url="https://api.london.cloud.service.gov.uk"
+  target_url=$(get_target_url)
+  if [[ "${target_url}" = "${prod_url}" ]] || [[ "${target_url}" = "${prod_london_url}" ]]; then
+    print_error "halting execution due to production protection"
+  fi
+}
+
+get_target_url() {
+  local target_url
+  target_url=$(jq -r '.Target' ~/.cf/config.json 2>&1)
+  if [[ "${target_url}" = "" ]]; then
+    print_error "could not determine target url"
+  fi
+  echo "${target_url}"
+}
+
+prepare_test() {
+  if [[ -d "${test_dir}" ]]; then
+    rm -rf "${test_dir}" >/dev/null 2>&1
+  fi
+  mkdir -p "${test_dir}" >/dev/null 2>&1
+}
+
+cleanup_test() {
+  rm -rf "${test_dir}" >/dev/null 2>&1
+}
+
+get_test_manifest() {
+  local name
+  name="${1}"
+  echo "applications:"
+  echo "- name: ${name}"
+  echo "  memory: 64M"
+  echo "  services:"
+  echo "  - ${name}"
+}
+
+#### test app/service using aws-s3-bucket service and staticfile buildpack
+deploy_test_app() {
+  local name
+  local dir
+  name="${1}"
+  dir="${test_dir}/${name}"
+  mkdir -p "${dir}"
+  get_test_manifest "${name}" >"${dir}/manifest.yml" 2>/dev/null
+  echo "${name}" >"${dir}/index.html" 2>/dev/null
+  touch "${dir}/Staticfile"
+  cf create-service -w aws-s3-bucket default "${name}" >/dev/null 2>&1
+  cd "${dir}" >/dev/null 2>&1
+  cf push "${name}" >/dev/null 2>&1
+}
+
+create_test_set() {
+  print_item "phase" "==== create_test_set ===="
+  local org_num
+  local org_name
+  local space_num
+  local space_name
+  local deploy_num
+  local deploy_name
+  for org_num in $(seq 1 "${test_org_count}"); do
+    org_name="${test_prefix}-org-${org_num}"
+    print_item "org" "${org_name}"
+    cf create-org "${org_name}" >/dev/null 2>&1
+    cf target -o "${org_name}" >/dev/null 2>&1
+    for space_num in $(seq 1 "${test_space_count}"); do
+      space_name="${org_name}-space-${space_num}"
+      print_item "space" "${space_name}"
+      cf create-space "${space_name}" >/dev/null 2>&1
+      cf target -s "${space_name}" >/dev/null 2>&1
+      for deploy_num in $(seq 1 "${test_deploy_count}"); do
+        deploy_name="${space_name}-app-${deploy_num}"
+        print_item "app/service" "${deploy_name}"
+        deploy_test_app "${deploy_name}" >/dev/null 2>&1
+      done
+    done
+  done
+}
+
+get_test_orgs() {
+  local org_num
+  local org_name
+  org_name=""
+  for org_num in $(seq 1 "${test_org_count}"); do
+    org_name="${org_name}${test_prefix}-org-${org_num},"
+  done
+  sed 's/,$//' < <(echo "${org_name}")
+}
```

Comment:
I don't think I'm a fan of mixing all the test code and functionality. You should consider a testing framework like bats. Here is a quick start guide, https://www.baeldung.com/linux/testing-bash-scripts-bats.

Reply:
will consider in future